### PR TITLE
fix(webrtc-star-transport/utils): properly handle legacy protocol edge case #369

### LINKED ISSUE #369: bug: make legacy protocol check more specific
I'm currently deploying the `star-signal` server via google cloud. I don't currently have a custom domain for it so the default generated domain is being used in my project. I named the deployment `libp2p-webrtc-star-signal` which means dns based multiaddrs look something like this: `/dnsaddr/libp2p-webrtc-star-signal.run.app/tcp/443/wss/p2p-webrtc-star`, where the domain is the auto-generated one provided by google cloud. This causes an issue in [cleanMultiaddr](https://github.com/libp2p/js-libp2p-webrtc-star/blob/590c5fc36a62631a0b0d102b56d60c091906c31c/src/utils.js#L28) because it doesn't validate where the legacy `/libp2p-webrtc-star` substring it located in the multiaddr.

A solution (which I'm taking) is to simply make sure `/libp2p-webrtc-star` doesn't appear anywhere else in your multiaddr. I don't think this is ideal due to it restricting the user on how they're able to name things. More importantly, it also requires users to be aware of the legacy `/libp2p-webrtc-star` in the first place. A much better solution is to validate that `/libp2p-webrtc-star` is being used as a protocol and not just some random substring found elsewhere in the multiaddr.

I'm planning to contribute the fix for this over the next day or so, as long as, no one else raises any problems with fixing it.

### PR DESCRIPTION
**Pull request by Zaba505**
Fixes #369

### CODE REVIEW

**Comment by Zaba505** [comment]
@mpetrunic Here's my fix, please review it whenever you have the time and thanks for reaching out. I honestly thought I had already contributed this fix but apparently forgot to. Whoops!

File: packages/webrtc-star-transport/src/utils.ts [hunk 1]
```diff
@@ -26,10 +26,10 @@ export function cleanUrlSIO (ma: Multiaddr) {
 }
 
 export function cleanMultiaddr (maStr: string) {
-  const legacy = '/libp2p-webrtc-star'
+  const legacy = '/libp2p-webrtc-star/'
 
   if (maStr.includes(legacy)) {
```
Comment (mpetrunic) [inline packages/webrtc-star-transport/src/utils.ts:31]: @achingbrain Would better fix be?
```suggestion
  if (maStr.startsWith(legacy)) {
```
I mean this works, so LGTM from me

Comment (mpetrunic) [reply packages/webrtc-star-transport/src/utils.ts:31]: Could you update PR to use this instead?

Comment (Zaba505) [reply packages/webrtc-star-transport/src/utils.ts]: @mpetrunic I've addressed your comment. Please let me know if anymore needs to be updated.

**Review by mpetrunic — APPROVED** [review]

**Pull request merged**